static/js/tools/map/chart_loader.tsx

```diff
@@ -133,6 +138,140 @@ export function ChartLoader(): JSX.Element {
   //   };
   // }, []);
 
+  // +++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++
+  // START useReducer
+  const [chartStore, dispatch] = useReducer(chartStoreReducer, emptyChartStore);
+  // END useReducer
+  // -------------------------------------------------------------------------
+
+  // +++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++
+  // The functions below are memoized and only change when the dependant
+  // variables change.
+  const geoJsonReady = useCallback(() => {
```

Comment:
does useCallback have to be in this file or can we separate it out into a different file that just holds all these callbacks?

Reply:
Looks like can create custom hook for this. moved them into ready_hook.ts now